service/src/integrationTest/kotlin/fi/espoo/evaka/decision/DecisionControllerIntegrationTest.kt

```diff
@@ -0,0 +1,288 @@
+// SPDX-FileCopyrightText: 2017-2024 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.decision
+
+import fi.espoo.evaka.FullApplicationTest
+import fi.espoo.evaka.application.ApplicationStateService
+import fi.espoo.evaka.application.ApplicationStatus
+import fi.espoo.evaka.application.ApplicationType
+import fi.espoo.evaka.application.DaycarePlacementPlan
+import fi.espoo.evaka.application.persistence.daycare.Adult
+import fi.espoo.evaka.application.persistence.daycare.Apply
+import fi.espoo.evaka.application.persistence.daycare.Child
+import fi.espoo.evaka.application.persistence.daycare.DaycareFormV0
+import fi.espoo.evaka.shared.ChildId
+import fi.espoo.evaka.shared.DaycareId
+import fi.espoo.evaka.shared.DecisionId
+import fi.espoo.evaka.shared.EmployeeId
+import fi.espoo.evaka.shared.PersonId
+import fi.espoo.evaka.shared.async.AsyncJob
+import fi.espoo.evaka.shared.async.AsyncJobRunner
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.auth.UserRole
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.dev.DevCareArea
+import fi.espoo.evaka.shared.dev.DevDaycare
+import fi.espoo.evaka.shared.dev.DevEmployee
+import fi.espoo.evaka.shared.dev.DevPerson
+import fi.espoo.evaka.shared.dev.DevPersonType
+import fi.espoo.evaka.shared.dev.insert
+import fi.espoo.evaka.shared.dev.insertTestApplication
+import fi.espoo.evaka.shared.dev.insertTestApplicationForm
+import fi.espoo.evaka.shared.domain.FiniteDateRange
+import fi.espoo.evaka.shared.domain.Forbidden
+import fi.espoo.evaka.shared.domain.MockEvakaClock
+import fi.espoo.evaka.shared.security.actionrule.AccessControlFilter
+import java.util.UUID
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.beans.factory.annotation.Autowired
+
+class DecisionControllerIntegrationTest : FullApplicationTest(resetDbBeforeEach = true) {
+    private val clock = MockEvakaClock(2024, 3, 1, 12, 0)
+
+    @Autowired lateinit var decisionController: DecisionController
+
+    @Autowired lateinit var applicationStateService: ApplicationStateService
+
+    @Autowired lateinit var asyncJobRunner: AsyncJobRunner<AsyncJob>
+
+    @Test
+    fun `PDF without contact info can be downloaded by service worker`() {
+        val serviceWorker =
```

Comment:
You can reduce some duplication by saving the DevEmployee, and converting it to a AuthenticatedUser directly:

```
val serviceWorker = DevEmployee(roles = setOf(UserRole.SERVICE_WORKER))
db.transaction { it.insert(serviceWorker) }
...
downloadPdf(serviceWorker.user, decisionId)
```

Basically there should almost never be a reason to instantiate AuthenticatedUser.Employee manually and duplicate the id and roles, because you're probably adding the same user to the db using DevEmployee. Also applies to citizen users, although you need to pass the auth level: `val citizenUser = somedevperson.user(CitizenAuthLevel.Strong)`

Reply:
ah nice! this was helpful in another PR too